src/sage/rings/polynomial/polynomial_ring_constructor.py

```diff
@@ -622,7 +625,10 @@ def PolynomialRing(base_ring, *args, **kwds):
         sage: R.<x,y> = PolynomialRing(RIF,2)
         sage: TestSuite(R).run(skip=['_test_elements', '_test_elements_eq_transitive'])
     """
-    if base_ring not in Rings():
+    is_tropical = False
+    if isinstance(base_ring, TropicalSemiring):
+        is_tropical = True
+    elif base_ring not in Rings():
```

Comment:
```suggestion
    if base_ring not in Rings() or not isinstance(base_ring, TropicalSemiring):
```
Just do the test a second time later on. It makes the logic a bit easier to follow IMO.